nan.h

```diff
@@ -1678,14 +1679,15 @@ class Callback {
   NAN_INLINE static void AsyncClose_(uv_handle_t* handle) {
     AsyncProgressWorker *worker =
             static_cast<AsyncProgressWorker*>(handle->data);
-    delete reinterpret_cast<uv_async_t*>(handle);
-    delete worker;
+    while(!worker->queue.empty()) {
+      delete worker->queue.front();
+      worker->queue.pop_front();
+    }
   }
```

Comment:
I think this introduces a memory leak.  What deletes the AsyncProgressWorker now?